consensus/istanbul/core/backlog_test.go

```diff
@@ -51,9 +51,46 @@ func TestCheckMessage(t *testing.T) {
 	testStates := []State{StateAcceptRequest, StatePreprepared, StatePrepared, StateCommitted, StateWaitingForNewRound}
 	testCode := []uint64{istanbul.MsgPreprepare, istanbul.MsgPrepare, istanbul.MsgCommit, istanbul.MsgRoundChange}
 
-	// future sequence
+	// accept Commits from previous sequence
```

Comment:
How hard would it be to test that we reject commits from the previous sequence if their digest doesn't match the latest block in our chain?

Reply:
We don't exactly reject the "commits", we're just initializing new roundstates with an empty parent commits field. 

There are no tests for `updateRoundState` available for this functionality currently. I can write some unit tests for it, but would require mocking everything (ie a mock `core`/`core.current`/`core.backend`) such that all execution paths in the `updateRoundState` function are explored.

What's hard is figuring out when this will actually happen with a real (ie non-mocked) chain, which I do not have an estimate for.